sys/kern/mmap.c

```diff
@@ -60,18 +60,29 @@ int do_munmap(vaddr_t addr, size_t length) {
 
   vm_map_t *uspace = proc_self()->p_uspace;
 
+  if (length == 0)
+    return EINVAL;
```

Comment:
I'm not sure, but according to NetBSD manual it is unnecessary. Please consult it with @cahirwpz.

Reply:
It looks good to me.
`EINVAL` is valid here for _FreeBSD_  and _Linux_ anyway if length is 0 then `munmap` has nothing to do.

Reply:
According to the POSIX:

```
The munmap() function shall fail if:
[...]
[EINVAL]
The len argument is 0.
```

https://pubs.opengroup.org/onlinepubs/9699919799/functions/munmap.html#